File: app/src/main/java/xyz/ratapp/ilx/ui/activities/MainActivity.java
package xyz.ratapp.ilx.ui.activities;

import android.content.res.Resources;
import android.os.Bundle;
import android.support.annotation.ColorInt;
import android.support.annotation.DrawableRes;
import android.support.annotation.IdRes;
import android.support.annotation.StyleRes;
import android.support.design.widget.NavigationView;
import android.support.v4.view.ViewPager;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.ActionBarDrawerToggle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.util.TypedValue;
import android.view.Gravity;
import android.view.View;
import android.widget.CompoundButton;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;

import xyz.ratapp.ilx.R;
import xyz.ratapp.ilx.controllers.main.MainController;
import xyz.ratapp.ilx.data.dao.User;
import xyz.ratapp.ilx.ui.helpers.ThemePicker;
import xyz.ratapp.ilx.ui.views.SlidingTabLayout;
import xyz.ratapp.ilx.ui.views.StatusSwitch;

public class MainActivity extends AppCompatActivity {

    //controller
    private MainController controller;
    //ui
    private ThemePicker picker;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        picker = new ThemePicker(this);
        setContentView(R.layout.activity_main);
        controller = new MainController(this);
    }

    public void setupUI() {
        //find views
        DrawerLayout layout = findViewById(R.id.dlMain);
        ViewPager container = findViewById(R.id.vpContainer);
        final Toolbar toolbar = findViewById(R.id.toolbar);
        final SlidingTabLayout slidingTabLayout = findViewById(R.id.stlTabs);
        final NavigationView navigationView = findViewById(R.id.navView);

        //toolbar
        toolbar.setTitle(R.string.requests);
        setSupportActionBar(toolbar);

        //tabs
        slidingTabLayout.setSelectedIndicatorColors(
                getResources().getColor(R.color.primary_dark_color));
        slidingTabLayout.setBackgroundResource(R.color.tab_bar_background_color);
        slidingTabLayout.setCustomTabView(R.layout.tab_layout, R.id.tvTabItem);
        slidingTabLayout.setDistributeEvenly(false);
        slidingTabLayout.setViewPager(container);

        //drawer
        ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(this, layout, toolbar,
                R.string.nav_drawer_open_desc, R.string.nav_drawer_close_desc);
        layout.addDrawerListener(toggle);
        toggle.syncState();

        //navigationView
        navigationView.setNavigationItemSelectedListener(controller);

        //status switch
        StatusSwitch status = new StatusSwitch(controller);
        toolbar.addView(status, new Toolbar.LayoutParams(Gravity.END));
    }

    public void bind(User user) {
        NavigationView navigationView = findViewById(R.id.navView);
        View layout = navigationView.getHeaderView(0);

        ImageView photo = layout.findViewById(R.id.ivNavIcon);
        TextView name = layout.findViewById(R.id.tvNavHeader);
        TextView text = layout.findViewById(R.id.tvNavText);

        Glide.with(this).load(user.getImage()).asBitmap().into(photo);
        name.setText(user.getFullName());
        text.setText(user.getStateText());

        picker.setTheme(user.isOnline());
    }

    @Override
    public void onBackPressed() {
        if (!controller.back()) {
            super.onBackPressed();
        }
    }

}

File: app/src/main/java/xyz/ratapp/ilx/ui/activities/MainActivity.kt
package xyz.ratapp.ilx.ui.activities

import android.os.Bundle
import android.support.v4.view.GravityCompat
import android.support.v7.app.ActionBarDrawerToggle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.view.Gravity
import android.view.Menu
import android.view.View
import android.widget.TextView
import com.bumptech.glide.Glide
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.app_bar_main.*
import kotlinx.android.synthetic.main.nav_header_main.view.*
import xyz.ratapp.ilx.R
import xyz.ratapp.ilx.controllers.main.MainController
import xyz.ratapp.ilx.data.dao.Uuser
import xyz.ratapp.ilx.ui.helpers.ThemePicker
import xyz.ratapp.ilx.ui.views.DrawerToggle
import xyz.ratapp.ilx.ui.views.StatusSwitch


class MainActivity : AppCompatActivity() {

    private var controller: MainController? = null
    private var picker: ThemePicker? = null
    private var status: StatusSwitch? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        controller = MainController(this)

        //setup status switch
        status = StatusSwitch(this)
        status!!.setController(controller)
        controller!!.setSwitch(status)
    }

    fun setupUI() {
        //tabs
        stlTabs.setSelectedIndicatorColors(resources.getColor(R.color.primary_dark_color))
        stlTabs.setBackgroundResource(R.color.tab_bar_background_color)
        stlTabs.setCustomTabView(R.layout.tab_layout, R.id.tvTabItem)
        stlTabs.setDistributeEvenly(false)

        picker = ThemePicker(this)
    }

    fun setupToolbar(title: String, customView: View) {
        //toolbar
        toolbar.title = title
        setSupportActionBar(toolbar)
        toolbar.addView(customView, Toolbar.LayoutParams(Gravity.END))

        //drawer
        val toggle = DrawerToggle(this, dlMain, toolbar,
                R.string.nav_drawer_open_desc, R.string.nav_drawer_close_desc)
        dlMain.addDrawerListener(toggle)
        toggle.syncState()

        //navigationView
        val navView = navView
        navView.setNavigationItemSelectedListener(controller)
        navView.setCheckedItem(R.id.nav_requests)
        navView.menu.findItem(R.id.nav_requests).title = title
    }

    fun bindUser(user: Uuser) {
        Glide.with(this).load(user.ava).error(R.mipmap.ic_launcher).
                into(navView.getHeaderView(0).ivNavIcon)
        navView.getHeaderView(0).tvNavHeader.text = user.courierName
        navView.getHeaderView(0).tvNavText.text = user.workStatus

        picker!!.setTheme(user.isOnline)
    }

    override fun onBackPressed() {
        if (dlMain.isDrawerOpen(GravityCompat.START)) {
            dlMain.closeDrawer(GravityCompat.START)
        }
        else {
            super.onBackPressed()
        }
    }

}
